query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+
+    Filter startTimeFilter =
+        createTimeFilter("Span.start_time_millis", Operator.GT, 1557780911508L);
+    Filter endTimeFilter = createTimeFilter("Span.end_time_millis", Operator.LT, 1557780938419L);
+
+    Filter andFilter =
+        Filter.newBuilder()
+            .setOperator(Operator.AND)
+            .addChildFilter(startTimeFilter)
+            .addChildFilter(endTimeFilter)
+            .build();
+    builder.setFilter(andFilter);
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "select encode(span_id, 'hex'), cast(tags as text), cast(request_headers as text) "
+            + "FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1557780911508 and end_time_millis < 1557780938419 )",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithoutFilter() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.addSelection(createColumnExpression("Span.id").build());
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQuerySingleDistinctSelection() {
+    Builder builder = QueryRequest.newBuilder();
+    builder.setDistinctSelections(true).addSelection(createColumnExpression("Span.id"));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select distinct encode(span_id, 'hex') FROM public.\"span-event-view\" "
+            + "where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "'",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryMultipleDistinctSelection() {
+    Builder builder = QueryRequest.newBuilder();
+    builder
+        .setDistinctSelections(true)
+        .addSelection(createColumnExpression("Span.id"))
+        .addSelection(createColumnExpression("Span.displaySpanName"))
+        .addSelection(createColumnExpression("Span.serviceName"));
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        builder.build(),
+        "Select distinct encode(span_id, 'hex'), span_name, service_name FROM public.\"span-event-view\" "
```

Comment:
In the case of multiple selections, do distinct apply to the first field?

Reply:
Distinct will apply to selection across multiple columns, and not just the first one.